### PR TITLE
Added a screen-blanking timeout feature.

### PR DESCRIPTION
**Pull request by henryso**
1. Added a screen blanking feature (which at the same time lowers the brightness to 0)
2. Added a `-timeout` flag to set the timeout (defaults to existing no-timeout behavior)
3. Added a `special` key action type
4. Added a `sleep` special action that invokes the blanking feature immediately

### CODE REVIEW

**Comment by muesli** [comment]
I think we should move the resume handling out of `triggerAction` and handle it directly from the `eventLoop`.

**Comment by henryso** [comment]
OK, I'll work on this.

File: widget_recent_window.go [hunk 1]
```diff
@@ -50,9 +50,6 @@ func (w *RecentWindowWidget) Update() error {
 	img := image.NewRGBA(image.Rect(0, 0, int(w.dev.Pixels), int(w.dev.Pixels)))
 
 	if int(w.window) < len(recentWindows) {
-		if w.lastID == recentWindows[w.window].ID {
```
Comment (muesli) [inline widget_recent_window.go:53]: Why did you remove these lines?

**Comment by henryso** [comment]
OK, I refactored the blanking code out of Deck.  I created a new DeviceWrapper to contain the logic and move the global variables into a struct.  Please let me know what I need to change.

File: widget_recent_window.go (hunk 1 above)
Comment (henryso) [reply widget_recent_window.go:53]: Hmm... I must've screwed that up.  Will fix.

Comment (henryso) [reply widget_recent_window.go:53]: OK, I remember.  This causes the force update *not* up update the recent windows when the device "wakes up" from sleep

Comment (henryso) [reply widget_recent_window.go:53]: Also, `RequiresUpdate` already gates the update on this, so I didn't think this would break things (and it doesn't seem to).

Comment (muesli) [reply widget_recent_window.go:53]: Ok, that makes sense!

File: README.md [hunk 2]
```diff
@@ -326,6 +332,22 @@ A list of available keycodes can be found here: [keycodes](https://github.com/mu
     value = "value"
 ```
 
+#### Switch to another deck
```
Comment (muesli) [inline README.md:335]: I know this can be a bit painful (and don't worry about it for now), but for the future: I'd appreciate documentation changes and things like the special actions in separate PRs. It keeps the entire review process a lot shorter and requires everyone to context switch less often.

Comment (henryso) [reply README.md:335]: OK. I'll try better in the future.  Is it OK to keep the documentation for the particular change in the same PR?

Comment (muesli) [reply README.md:335]: Yes, don't worry, I don't want to cause more work on your side for now! Great work!

**Comment by muesli** [comment]
> OK, I refactored the blanking code out of Deck. I created a new DeviceWrapper to contain the logic and move the global variables into a struct. Please let me know what I need to change.

This is a great idea, but got me thinking: maybe that's universal functionality that should actually go to upstream `streamdeck` library: https://github.com/muesli/streamdeck

I'm having a look at the API right now and this might integrate well. I might simply move some of your code over there in a few minutes.

**Comment by muesli** [comment]
> I'm having a look at the API right now and this might integrate well. I might simply move some of your code over there in a few minutes.

Ok, this took a little longer than anticipated, but I think I've basically cracked this :smile: Will push a few cleanups here afterwards and then we can finally start merging the rest of your fixes!

**Comment by muesli** [comment]
Sleep timeout feature PR for upstream streamdeck: https://github.com/muesli/streamdeck/pull/8

**Comment by muesli** [comment]
I've added a sleep timeout flag in #68. We can cherry-pick the special action stuff from this PR once that's merged.

**Comment by henryso** [comment]
OK, I'll give this a shot.  If I mess up the branches too much, I'll close this and create a new PR.

**Comment by muesli** [comment]
@henryso Don't worry, you don't have to touch / push anything, I can rebase that for you rather easily here.

**Comment by henryso** [comment]
OK, If you want me to do something here, let me know and I'll try my best.

**Comment by muesli** [comment]
Instead of rebasing I decided to simply cherry-pick the remaining change from here into #72. Thank you @henryso!
